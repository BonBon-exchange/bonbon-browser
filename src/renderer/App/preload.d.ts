declare global {
  interface Window {
    app: {
      analytics: {
        event: (eventName: string, params?: Record<string, string>) => void;
      };
      tools: {
        inspectElement: (point: { x: number; y: number }) => void;
      };
      board: {
        open: (board: {
          id: string;
          label: string;
          isFullSize: boolean;
        }) => void;
        close: () => void;
        selectNext: () => void;
      };
      browser: {
        select: (webContentsId: string) => void;
        selectBrowserView: () => void;
      };
      listener: {
        newWindow: (action: unknown) => void;
        loadBoard: (action: unknown) => void;
        purge: (action: unknown) => void;
        showLibrary: (action: unknown) => void;
        showSettings: (action: unknown) => void;
        saveBoard: (action: unknown) => void;
        renameBoard: (action: unknown) => void;
        closeWebview: (action: unknown) => void;
        closeAllWebview: (action: unknown) => void;
      };
      off: {
        newWindow: () => void;
        loadBoard: () => void;
        showLibrary: () => void;
        showSettings: () => void;
        purge: () => void;
        saveBoard: () => void;
        renameBoard: () => void;
        closeWebview: () => void;
        closeAllWebview: () => void;
      };
    };
    darkMode: {
      toggle: () => void;
      system: () => void;
    };
  }
}

export {};
