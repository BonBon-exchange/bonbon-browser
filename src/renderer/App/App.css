/*
 * @NOTE: Prepend a `~` to css file paths that are in your node_modules
 *        See https://github.com/webpack-contrib/sass-loader#imports
 */
body {
  font-family: sans-serif;
}

.bold {
  font-weight: bold;
}
