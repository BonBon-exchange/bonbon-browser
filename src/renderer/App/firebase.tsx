// src/firebase.ts

import { initializeApp } from 'firebase/app';
import { getDatabase } from 'firebase/database';
import { getAuth, signInAnonymously } from 'firebase/auth';
import { getAnalytics, logEvent } from 'firebase/analytics';

// Your Firebase configuration (replace with your actual config)
const firebaseConfig = {
  apiKey: 'AIzaSyBZ0V6V43EW3NHE-z0fZZnlpe22mE55uqQ', // Replace with your API key
  authDomain: 'bonbon-browser.firebaseapp.com', // Replace with your Auth domain
  databaseURL:
    'https://bonbon-browser-default-rtdb.europe-west1.firebasedatabase.app', // Replace with your Database URL
  projectId: 'bonbon-browser', // Replace with your Project ID
  storageBucket: 'bonbon-browser.appspot.com', // Replace with your Storage Bucket
  messagingSenderId: '313574192651', // Replace with your Messaging Sender ID
  appId: '1:313574192651:web:cbe8cd4ce4797d7c82f90e', // Replace with your App ID
  measurementId: 'G-GVME8GLMGZ',
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);

// Initialize Realtime Database and Auth
const database = getDatabase(app);
const auth = getAuth(app);

const analytics = getAnalytics();

// Sign in anonymously
signInAnonymously(auth)
  .then(() => {
    console.log('Signed in anonymously to Firebase Auth');
    logEvent(analytics, 'firebase_signin_anonymous');
  })
  .catch((error) => {
    console.error('Firebase Auth Error:', error);
    logEvent(analytics, 'firebase_signin_anonymous_error');
  });

export { database, analytics };
