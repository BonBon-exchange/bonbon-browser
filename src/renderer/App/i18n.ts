import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import en from 'renderer/App/locales/en.json';
import es from 'renderer/App/locales/es.json';
import fr from 'renderer/App/locales/fr.json';
import nl from 'renderer/App/locales/nl.json';
import pl from 'renderer/App/locales/pl.json';
import ru from 'renderer/App/locales/ru.json';
import ar from 'renderer/App/locales/ar.json';
import cn from 'renderer/App/locales/cn.json';
import pt from 'renderer/App/locales/pt.json';
import tr from 'renderer/App/locales/tr.json';
import de from 'renderer/App/locales/de.json';
import fa from 'renderer/App/locales/fa.json';
import ja from 'renderer/App/locales/ja.json';

const resources = {
  en: {
    translation: en,
  },
  es: {
    translation: es,
  },
  fr: {
    translation: fr,
  },
  nl: {
    translation: nl,
  },
  pl: {
    translation: pl,
  },
  ru: {
    translation: ru,
  },
  ar: {
    translation: ar,
  },
  cn: {
    translation: cn,
  },
  pt: {
    translation: pt,
  },
  tr: {
    translation: tr,
  },
  de: {
    translation: de,
  },
  fa: {
    translation: fa,
  },
  ja: {
    translation: ja,
  },
};

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    resources,
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
