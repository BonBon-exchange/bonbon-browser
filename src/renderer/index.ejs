<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <meta
      http-equiv="Content-Security-Policy"
    />
    <title>BonBon</title>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>
