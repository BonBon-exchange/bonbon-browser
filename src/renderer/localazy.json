{
  "writeKey": "a8141754324130152091-47656f56226b2301495b4076811b79be71789e8713f0656c8520b349fe095e32",
  "readKey": "a8141754324130152091-5b6ed7071b7e42b0ed06f22256b6b6381802977416d00c484d8e81f4f1ba2ef0",

  "upload": {
    "type": "json",
    "files": "App/locales/en.json"
  },

  "download": {
    "files": "App/locales/${lang}.json"
  }
}
