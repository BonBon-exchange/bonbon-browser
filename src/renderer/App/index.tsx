/* eslint-disable camelcase */
import { createRoot } from 'react-dom/client';

import { ChatState } from 'types/chat';
import { IpcRendererEvent } from 'electron';
import { Expected } from '-lola/sepyt/lola';
import ManagedMemoryProxy from '-lola/seludom/mm';
import { App } from './App';

(window as Expected)['----/__/0/CDXX/-'] = {
  ':.BonBon': {
    chat: { username: '', isMagic: false },
  },
} as Expected;

const bonbon_state: { chat: ChatState } = (window as Expected)[
  '----/__/0/CDXX/-'
][':.BonBon'];
['states'];

const bonboot = ({
  context,
}: {
  context: {
    states: {
      chat: ChatState;
    };
  };
}) => {
  try {
    bonbon_state.chat = context.states.chat;
    
    const container = document.getElementById('root')!;
    const root = createRoot(container);
    const handleChatState = (
      _e: IpcRendererEvent,
      args: {
        chatState: ChatState;
      }
    ) => {
      bonbon_state.chat = args.chatState;
      root.render(<App chatState={bonbon_state.chat} />);
    };

    window.app.listener.chatState(handleChatState);
    root.render(<App chatState={bonbon_state.chat} />);
  } catch (e) {
    window.app.off.chatState();
  }
};

const BonBonMMP = new ManagedMemoryProxy(() => ({
  bonboot: (bootArgs: Expected) => bonboot(bootArgs),
}));

console.log({ bonboot });

const bbproxy = BonBonMMP.getProxy();

(global as Expected).bonboot = bonboot;

console.log({ bbproxy });

bbproxy().bonboot({ context: { states: bonbon_state } });
export default BonBonMMP;
