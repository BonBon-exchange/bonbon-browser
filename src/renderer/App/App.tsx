/* eslint-disable import/prefer-default-export */
import { useState, useRef, useCallback, useEffect } from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';

import { store, getPersistedStoreAndPersistor } from 'renderer/App/store/store';
import { Addaps } from 'renderer/App/components/Addaps';

import './i18n';

import 'renderer/App/style/dark.css';
import 'renderer/App/style/light.css';
import './App.css';

export function App() {
  const [isLoadedBoard, setIsLoadedBoard] = useState<boolean | string>(false);
  const [boardId, setBoardId] = useState<string>('');
  const persisted = useRef<any>(null);

  const loadBoardAction = useCallback(
    (_e: any, args: { boardId: string }) => {
      if (args.boardId === boardId) return;
      persisted.current = getPersistedStoreAndPersistor(args.boardId);
      setBoardId(args.boardId);
      setIsLoadedBoard(true);
    },
    [boardId]
  );

  const purgeAction = useCallback(() => {
    if (persisted.current?.persistor) {
      persisted.current?.persistor.purge();
      localStorage.removeItem(`persist:${boardId}`);
    }
  }, [boardId]);

  useEffect(() => {
    window.app.listener.loadBoard(loadBoardAction);
    return () => window.app.off.loadBoard();
  }, [loadBoardAction]);

  useEffect(() => {
    window.app.listener.purge(purgeAction);
    return () => window.app.off.purge();
  }, [purgeAction]);

  useEffect(() => {
    window.app.analytics.event('load_app');
  }, []);

  useEffect(() => {
    window.addEventListener('contextmenu', (e) => {
      e.preventDefault();
      const target = e.target as HTMLDivElement;

      if (
        document
          .getElementById('LeftBar__browserFavContainerItems')
          ?.contains(target)
      )
        window.app.tools.showLeftbarContextMenu({ x: e.clientX, y: e.clientY });
    });
  }, []);

  return isLoadedBoard ? (
    <Provider store={persisted.current?.store}>
      <PersistGate loading={null} persistor={persisted.current?.persistor}>
        <Addaps boardId={boardId} />
      </PersistGate>
    </Provider>
  ) : (
    <Provider store={store}>
      <Addaps />
    </Provider>
  );
}
