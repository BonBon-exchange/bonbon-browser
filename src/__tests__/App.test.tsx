import '@testing-library/jest-dom';
import { act } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { mockWindow } from './beforeAll';
import { App } from '../renderer/App/App';

let tree: any;

describe('App', () => {
  beforeAll(() => {
    mockWindow();
  });

  it('should render', () => {
    act(() => {
      tree = renderer.create(<App />);
    });
    expect(tree).toMatchSnapshot();
  });
});
