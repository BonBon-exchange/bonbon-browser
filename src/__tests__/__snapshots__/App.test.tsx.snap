// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div
  className="Loader__loading"
>
  <img
    src="test-file-stub"
  />
</div>
`;
